pkg/controller/machineset/controller.go

```diff
@@ -188,12 +189,13 @@ func (r *ReconcileMachineSet) Reconcile(ctx context.Context, request reconcile.R
 				"The AuthoritativeAPI is set to %s", string(machineSet.Status.AuthoritativeAPI),
 			))
 
-			machineSet, err := updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
+			_, err := updateMachineSetStatus(r.Client, machineSet, machineSetCopy.Status)
```

Comment:
Not updating the `machineSet` in place here is fine, as if we're in this conditional we will return and do no further processing anyway.